sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkHttpClientBuilder.java

```diff
@@ -45,17 +57,30 @@ public class JdkHttpClientBuilder {
     static final Set<String> DEFAULT_RESTRICTED_HEADERS;
 
     static {
+        Configuration configuration = Configuration.getGlobalConfiguration();
+
+        DEFAULT_CONNECTION_TIMEOUT = getDefaultTimeoutFromEnvironment(configuration,
```

Comment:
is there any way we can do parsing centrally and not inside the client implementations?

Reply:
Yeah, I think this can get cleaned up. Will do that after this round of CI runs to make sure there aren't more fixes to be done.

Reply:
You know, I think this can be done by newing up an `HttpClientOptions`. I think it does this as well

Reply:
I see this PR does not change status quo, so I created #39456 to follow up